src/room.ts

```diff
@@ -209,23 +231,47 @@ export class Room extends EventEmitter implements Sayable {
     return rawMemberList.map(m => Contact.load(m.UserName))
   }
 
+  /**
+   * ISSUE #64 emoji need to be striped
+   * ISSUE #104 never use remark name because sys group message will never use that
+   * @rui: cannot use argument NickName because it mix real nick and remark 
+   */
   private parseNickMap(memberList: RoomRawMember[]): Map<string, string> {
     const nickMap: Map<string, string> = new Map<string, string>()
-
+    let contact: Contact
     if (memberList && memberList.map) {
       memberList.forEach(member => {
-        /**
-         * ISSUE #64 emoji need to be striped
-         * ISSUE #104 never use remark name because sys group message will never use that
-         */
-        nickMap[member.UserName] = UtilLib.stripEmoji(
-          member.DisplayName || member.NickName
-        )
+        contact = Contact.load(member.UserName)
+        nickMap[member.UserName] = UtilLib.stripEmoji(contact.name())
       })
     }
     return nickMap
   }
 
+  private parseRemarkMap(memberList: RoomRawMember[]): Map<string, string> {
+    const remarkMap: Map<string, string> = new Map<string, string>()
+    let contact: Contact
+    let remark: string | null
+    if (memberList && memberList.map) {
+      memberList.forEach(member => {
+        contact = Contact.load(member.UserName)
+        remark =  contact.remark() || ''
+        remarkMap[member.UserName] = UtilLib.stripEmoji(remark)
+      })
+    }
+    return remarkMap
+  }
+
+  private parseDisplayName(memberList: RoomRawMember[]): Map<string, string> {
```

Comment:
> parseRemarkMap & parseDisplayName has many duplicated code.

> /**
        * ISSUE #64 emoji need to be striped
         * ISSUE #104 never use remark name because sys group message will never use that
         */

Actually, parseRemarkMap & parseDisplayName & parseNickMap has many duplicated code.   So should I not duplicated all?  Then maybe I cannot keep the original comments in the code...